OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/ImportKeysActivity.java

```diff
@@ -113,6 +117,12 @@ protected void initLayout() {
         setContentView(R.layout.import_keys_activity);
     }
 
+    @Override
+    public void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        handleActions(intent);
```

Comment:
there *is* `setIntent()` in Activity, I feel like the clean solution here is to set the activity's intent to the new one and call handleActions (which mostly manages fragment state) in `onResumeFragments` any time the intent changed. this also puts it in the correct lifecycle for the fragment manager, which is not yet initialized during `onNewIntent` (or `onResume`) but is initialized in `onResumeFragments`

see also [FragmentActivity](http://developer.android.com/reference/android/support/v4/app/FragmentActivity.html#onResume()) and [commit](http://developer.android.com/reference/android/app/FragmentTransaction.html#commit()) android docs